bitcoind-rpc/src/main/scala/org/bitcoins/rpc/config/BitcoindConfig.scala

```diff
@@ -0,0 +1,323 @@
+    val startIndexOpt = lines.indexOf(s"[$ourNetworkString]") match {
+      case -1 =>
+        None
+      case other: Int => Some(other)
+    }
+
+    for {
+      startIndex <- startIndexOpt
+      endIndex = {
+        // we're looking for the first section _after_ the
+        // section for our current network
+        val toSearchIn = lines.zipWithIndex.drop(startIndex)
+        toSearchIn
+          .find {
+            case (line, _) =>
+              otherNetworks.exists(net => s"[$net]" == line)
+          }
+          .map {
+            case (_, index) => index
+          }
+          // we got to the end without finding a new section header,
+          // that means we can search to the bottom
+          .getOrElse(lines.length)
+      }
+      result <- {
+        val linesToSearchIn = lines.slice(startIndex, endIndex)
+        val collect = collectFrom(linesToSearchIn)(_)
+        collect {
+          case (`key`, value) => value
+        }.headOption
+      }
+    } yield result
+  }
+
+  /**
+    * Searches the config for a key matchinng the provided
+    * string that's _not_ in a section header or prefixed
+    * by a network.
+    */
+  private def readRawOpt(key: String): Option[String] = {
+    val linesToSearchIn = lines.takeWhile(!_.startsWith("["))
+    val collect = collectFrom(linesToSearchIn)(_)
+    collect {
+      case (`key`, value) => value
+    }.headOption
+  }
+
+  lazy val datadir: Option[File] = getValue("datadir").map(new File(_))
+
+  lazy val username: Option[String] = getValue("rpcuser")
+  lazy val password: Option[String] = getValue("rpcpassword")
+  lazy val zmqpubrawblock: Option[URI] =
+    getValue("zmqpubrawblock").map(new URI(_))
+  lazy val zmqpubrawtx: Option[URI] = getValue("zmqpubrawtx").map(new URI(_))
+  lazy val zmqpubhashblock: Option[URI] =
+    getValue("zmqpubhashblock").map(new URI(_))
+  lazy val zmqpubhashtx: Option[URI] = getValue("zmqpubhashtx").map(new URI(_))
+
+  lazy val port: Int = getValue("port").map(_.toInt).getOrElse(network.port)
+
+  /** Defaults to localhost */
+  lazy val bind: URI = new URI({
+    val baseUrl = getValue("bind").getOrElse("localhost")
+    if (baseUrl.startsWith("http")) baseUrl
+    else "http://" + baseUrl
+
+  })
+
+  lazy val uri: URI = new URI(s"$bind:$port")
+
+  lazy val rpcport: Int =
+    getValue("rpcport").map(_.toInt).getOrElse(network.rpcPort)
+
+  /** Defaults to localhost */
+  lazy val rpcbind: URI = new URI({
+    val baseUrl = getValue("rpcbind").getOrElse("localhost")
+    if (baseUrl.startsWith("http")) baseUrl
+    else "http://" + baseUrl
+  })
+  lazy val rpcUri: URI = new URI(s"$rpcbind:$rpcport")
+
+  /** Creates a new config with the given keys and values appended */
+  def withOption(key: String, value: String): BitcoindConfig = {
+    val ourLines = this.lines
+    new BitcoindConfig {
+
+      def lines: Seq[String] = {
+        val newLine = s"$key=$value"
+        Seq(newLine) ++ ourLines
```

Comment:
just a note, you don't need to wrap in `Seq(newLine)` you can do `newLine +: ourLines`.